### DISCUSSION TITLE
Effect Preset Events

### DISCUSSION THREAD

**Discussion by Jaguarious** [oddbear/GoXLR-TouchPortal-Plugin#27] [Ideas]
This plugin shows some real promise.  I think it is really worth further development.  I love the idea as a whole.

One thing that would really make it useful for me as a streamer - would be triggering one of the six FX presets.   I mean - profile switching is nice.  But being able to trigger a voice changing effect would be a far more valuable tool for a live streamer.

**Comment by oddbear** [comment]
Since there is no officially support for Touch Portal by TC-Helicon, this plugin makes the GoXLR App think it's a StreamDeck. Therefor it's also limited to what the StreamDeck plugin can do.

This feature has been been asked for before a couple of times on the TC-Helicon discord in the `new-feature-ideas-and-requests` channel, and that is the right place to ask for updates on the StreamDeck plugin (they do listen and prioritize the most common requests).

There's also a lot more features I would like them to add, like get volume, set volume, get mute, set mute.

From my understanding, most or all this things are actually done through their app already, ex. pressing mute, will send a message to the app, that sends two messages back to the device, one for changing the light on the button, and one for actually muting. Because of this the buttons will not work if the app is running.
That also means that everything that stops working if you close the app, or can already be managed trough the app, should be easy for them to implement in their StreamDeck plugin if they want.
